### PR TITLE
Support HTTP / HTTPS agent

### PR DESCRIPTION
**Pull request by yuxhuang**
There are a few use cases of using a HTTP or HTTPS agent, including:
- Use a custom pool of sockets
- Allow use of self-signed certificates in HTTPS connections
- Add SSL client certificates and private keys

The certificate use is a very useful scenario when doing development or connecting to a self-signed HTTPS server.

References:
HTTP Agent: http://nodejs.org/api/http.html#http_class_http_agent
HTTPS Agent: http://nodejs.org/api/https.html#https_class_https_agent

### CODE REVIEW

File: http.js [hunk 1]
```diff
@@ -280,7 +280,8 @@ exports.request = function (request) {
             "port": request.port || (ssl ? 443 : 80),
             "path": request.path || "/",
             "method": request.method || "GET",
-            "headers": headers
+            "headers": headers,
+            "agent": request.agent || undefined
```
Comment (kriskowal) [inline http.js]: `|| undefined` is superfluous.

**Comment by kriskowal** [comment]
Thanks for the PR. I’m glad it’s this simple.

**Comment by yuxhuang** [comment]
Thanks for your quick response. I removed "undefined" in the line. As you pointed out, it could be actually a bug (since "false" is allowed in this option.)

**Comment by kriskowal** [comment]
Thanks.

**Comment by kriskowal** [comment]
If you get a chance to write specs and update the README, that’d help the next user as well. Otherwise, I’ll try to pick it up next time I do a release.

**Comment by yuxhuang** [comment]
Yes, I think I will pick it up and send a PR later this week.

**Pull request merged**